fsck/fsck.c

```diff
@@ -732,31 +741,31 @@ static int read_boot_region(struct exfat_blk_dev *bd, struct pbr **pbr,
 	return ret;
 }
 
-static int restore_boot_region(struct exfat_blk_dev *bd)
+static int restore_boot_region(struct exfat_blk_dev *bd, unsigned int sect_size)
 {
 	int i;
 	char *sector;
 	int ret;
 
-	sector = malloc(bd->sector_size);
+	sector = malloc(sect_size);
 	if (!sector)
 		return -ENOMEM;
 
 	for (i = 0; i < 12; i++) {
-		if (exfat_read(bd->dev_fd, sector, bd->sector_size,
-				BACKUP_BOOT_SEC_IDX * bd->sector_size +
-				i * bd->sector_size) !=
-				(ssize_t)bd->sector_size) {
+		if (exfat_read(bd->dev_fd, sector, sect_size,
```

Comment:
I still think this is weird. If you update bd->sector_size on caller, There is no need to make unnecessary changes.

Reply:
this function could be called many times to read the backup boot region. I think sect_size parameter is better than updating bd->sector_size for each call. And in fsck, bd->sector_size is not referenced anymore except this function.